CMakeLists.txt

```diff
@@ -111,14 +111,17 @@ endmacro()
 
 # Skip building yaml if the expected version is already present in the system
 if(yaml_FOUND)
-  if("${yaml_VERSION}" VERSION_EQUAL 0.2.5)
+  if("${yaml_VERSION}" VERSION_EQUAL 0.2.2)
     set(_SKIP_YAML_BUILD 1)
+  # Build yaml if the system version is greater than 0.2.2
+  elseif("${yaml_VERSION}" VERSION_GREATER 0.2.2)
+    set(_SKIP_YAML_BUILD 0)
```

Comment:
We discussed this in the Waffle meeting today.  The general feeling was that it is OK to enforce a maximum, but we probably shouldn't do it so strictly.  For instance, our rosdep keys don't make this enforcement.  So the suggestion here is to only build locally if the system version is greater than the next minor version, i.e. this check should be `>= 0.3.0`.